### ISSUE TITLE
[feature-request] crop support

### ISSUE THREAD

**Issue by Ulexus** [jtheoof/swappy#55]
While you can preselect with slurp, there are situations where more refinement is useful.  A cropping tool would be most handy.

**Comment by cauebs** [comment]
Just to add to what @Ulexus said, sometimes you want to capture something that requires a mouse hover, and thus using slurp is not viable, requiring you to capture the entire screen. Being able to capture first and crop later would be great (like what [flameshot](https://github.com/flameshot-org/flameshot/) does). Today I have to resort to GIMP for this quite often.

**Comment by foonathan** [comment]
I'd particularly like to see an "auto crop" tool, since doing a preselection isn't precise. I can do a PR if you're interested.

**Comment by matthewarmand** [comment]
> I found a workaround if you use `grim` and `slurp`. You can take a full screenshot, when swappy opens, make a screenshot of swappy window. That way you'll have two instances of swappy and second one will have cropped image.

Yeah I've been doing this too, but it does sacrifice some resolution on the ultimate image. I think a crop feature in swappy would still be preferable.

**Comment by adonespitogo** [comment]
This feature is really nifty for any annotation tool.

**Comment by jtheoof** [comment]
I agree, I'm hopping to get back to it at some point but my time on the project is currently very limited due to family reasons. It will come out at some point.

**Comment by paulpet** [comment]
Just adding my support for a cropping feature to be added. It was one of the first features I went to test upon first opening swappy. :(

**Comment by touero** [comment]
So, this feature will be supported?
